msal/application.py

```diff
@@ -69,6 +70,36 @@ def _clean_up(result):
     return result
 
 
+def _preferred_browser():
+    """Register Edge and return a name suitable for webbrowser.get(...),
+    if running on Linux and there is no BROWSER env var, otherwise return None.
+    """
+    browser_path = "/usr/bin/microsoft-edge"  # Use a full path owned by sys admin
+    user_has_no_preference = "BROWSER" not in os.environ
+    user_opted_in_for_edge = "microsoft-edge" in os.environ.get("BROWSER", "")
+    if ((user_opted_in_for_edge or user_has_no_preference)
```

Comment:
To my surprise, based on my hack-and-learn-webbrowser experience, a `BROWSER=microsoft-edge` would not work, due to [this issue I reported to Python](https://bugs.python.org/issue44844) and [reported to Edge](https://microsoft.visualstudio.com/Edge/_workitems/edit/35081718?src=WorkItemMention&src-action=artifact_link). The workaround is to launch Edge by `BackgroundBrowser`, and the `user_opted_in_for_edge` logic here is to shield those users who opted in for Edge from this issue.

Perhaps I should also document this history into the source code?

Reply:
That assumes users who registered `microsoft-edge` and set `BROWSER=microsoft-edge` didn't use `BackgroundBrowser`. What if they did, and made their own registration because they needed to specify different path?

Reply:
> That assumes users who registered `microsoft-edge` and set `BROWSER=microsoft-edge` didn't use `BackgroundBrowser`. What if they did, and made their own registration because they needed to specify different path?

By "users who registered `microsoft-edge`", did you mean they did `webbrowser.register("microsoft-edge", ...)`? Those users would not be affected by this PR, because MSAL registers its own with the name `msal-edge`.

Reply:
Those users are affected in that MSAL would override their configuration. If the user does `webbrowser.register("microsoft-edge", ...)` and `BROWSER=microsoft-edge`, MSAL registers msal-edge and launches that instead.

Reply:
Concluding this conversation. The next commit will contain more comments based on the recent conversation.